internal/services/datafactory/data_factory_linked_service_azure_blob_storage_resource.go

```diff
@@ -23,8 +23,10 @@ func resourceDataFactoryLinkedServiceAzureBlobStorage() *pluginsdk.Resource {
 		Update: resourceDataFactoryLinkedServiceBlobStorageCreateUpdate,
 		Delete: resourceDataFactoryLinkedServiceBlobStorageDelete,
 
-		// TODO: replace this with an importer which validates the ID during import
-		Importer: pluginsdk.DefaultImporter(),
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.LinkedServiceID(id)
+			return err
```

Comment:
could we add a custom importer for linked services that also checks the types here?

Reply:
done! 🙂